packages/@react-stately/slider/src/useSliderState.ts

```diff
@@ -192,9 +192,11 @@ export function useSliderState<T extends number | number[]>(props: SliderStateOp
   const [focusedIndex, setFocusedIndex] = useState<number | undefined>(undefined);
 
   const valuesRef = useRef<number[]>(null);
-  valuesRef.current = values;
   const isDraggingsRef = useRef<boolean[]>(null);
-  isDraggingsRef.current = isDraggings;
+  useEffect(() => {
```

Comment:
can't use this https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/utils/src/useLayoutEffect.ts

Reply:
It turned out that we don't even need to assign these refs during render. We only need them so that we can fire `onChangeEnd` immediately after `onChange` without waiting for a render in between. If we always assign them at the same time we update the state it seems to work. Done in #4564.